specification/purview/data-plane/Azure.Analytics.Purview.Insights/preview/2022-03-01-preview/purviewinsights.json

```diff
@@ -0,0 +1,1202 @@
+        "operationId": "ScanStatus_SourceTypes",
+        "produces": ["application/json"],
+        "parameters": [
+          {
+            "in": "query",
+            "name": "collectionId",
+            "type": "string"
+          },
+          {
+            "in": "query",
+            "name": "window",
+            "type": "integer",
+            "format": "int32",
+            "default": 0
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/ScanSourceTypesResponse"
+            }
+          },
+          "default": {
+            "description": "An error response received from the Reporting Service",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseModel"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ScanStatus_SourceTypes": {
+            "$ref": "./examples/ScanStatus_SourceTypes.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SnapshotRecord": {
+      "required": ["kind"],
+      "type": "object",
+      "properties": {
+        "kind": {
+          "enum": [
+            "AssetFilesAggregation",
+            "AssetDistribution",
+            "FilesWithoutResourceSet",
+            "AllAssetsDistribution",
+            "AssetCurationDistribution",
+            "DeletedAssetsDistribution",
+            "NetNewAssetsDistribution",
+            "Tags"
+          ],
+          "type": "string"
+        },
+        "snapshotTime": {
+          "format": "date-time",
+          "type": "string"
+        }
+      },
+      "discriminator": "kind"
+    },
+    "Snapshot": {
+      "type": "object",
+      "properties": {
+        "snapshotTime": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SnapshotRecord"
+          }
+        },
+        "count": {
+          "format": "int32",
+          "type": "integer"
+        }
+      }
+    },
+    "ErrorModel": {
+      "type": "object",
+      "properties": {
+        "errorCode": {
+          "type": "string",
+          "readOnly": true
+        },
+        "message": {
+          "type": "string",
+          "readOnly": true
+        },
+        "exceptionCulprit": {
+          "type": "string",
+          "readOnly": true
+        },
+        "target": {
+          "type": "string",
+          "readOnly": true
+        },
+        "details": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ErrorModel"
+          },
+          "readOnly": true
+        }
+      }
+    },
+    "ErrorResponseModel": {
+      "type": "object",
+      "properties": {
+        "error": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/ErrorModel"
+            }
+          ]
```

Comment:
[Purview Scanning API](https://github.com/Azure/azure-rest-api-specs/blob/5dad3b83d0926768ede15ed41895d5f38defd512/specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2021-10-01-preview/scanningService.json#L1907) is specified the same. The ErrorResponseModel is an object containing the Error. 
Follows [Microsoft REST API Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#errorresponse--object)

Reply:
Yes ... it is wrong there too.  Please fix.

Reply:
I'm trying to sync this with our other services, the same thing on our [RP](https://github.com/Azure/azure-rest-api-specs/blob/4bd83868c6ae7f731874b6673f400ca97914ee97/specification/purview/resource-manager/Microsoft.Purview/stable/2021-07-01/purview.json#L1395) which is in stable release as well. I assume this would be some change in how autorest works. Can you point toward what would be needed?

Reply:
You can just accept the suggestion I made in the original comment.  That is what is needed.

Reply:
Can you reference how I make this change via the swagger operations available in C#? Every change is a generated change, not a manual update to the swagger spec. If you're not familiar, perhaps the review board knows this and I'll sync up with them. I haven't yet found a reference to how I can implement your suggestion; I understand that is what is correct per the guideline.

Reply:
Figured this out. Autorest v3 does it the suggested way. We were using v2 and I got the migration to work for us.